gradunwarp/core/unwarp_resample.py

```diff
@@ -314,10 +308,11 @@ def non_linear_unwarp_siemens(self, volshape, dv, dxyz, m_rcs2lai, m_rcs2lai_noh
             out[..., s] = im2
 
         print()
-       
-        img=nib.Nifti1Image(fullWarp,self.m_rcs2ras)
+
+        img=nib.Nifti1Image(fullWarp, self.m_rcs2ras)
         nib.save(img,"fullWarp_abs.nii.gz")
         # return image and the jacobian
+        del vrcsw, vfsl, vxyzw, vrcs, vxyz, vrcsg, fullWarp
```

Comment:
This looks like it may be deleting local variables immediately before a function return (which should cause all local variables to be deleted anyway, right?).  Why is this necessary?

Reply:
Remains of function based memory profiling, indeed unnecessary.